support-workers/src/typescript/lambdas/createPaymentMethodLambda.ts

```diff
@@ -0,0 +1,194 @@
+import { getStripeConfig, StripeService } from '../services/stripe';
+import { getIfDefined } from '../util/nullAndUndefined';
+
+const stage = stageFromEnvironment();
+const stripeServiceHandler = new ServiceHandler(stage, async (stage) => {
+	const config = await getStripeConfig(stage);
+	return new StripeService(config);
+});
+const paypalServiceHandler = new ServiceHandler(stage, async (stage) => {
+	const config = await getPayPalConfig(stage);
+	return new PayPalService(config);
+});
+
+export const handler = async (
+	state: WrappedState<CreatePaymentMethodState>,
+) => {
+	console.info(`Input is ${JSON.stringify(state)}`);
+	const createPaymentMethodState = wrapperSchemaForState(
+		createPaymentMethodStateSchema,
+	).parse(state).state;
+	return createSalesforceContactState(
+		createPaymentMethodState,
+		await createPaymentMethod(
+			createPaymentMethodState.paymentFields,
+			createPaymentMethodState.user,
+		),
+	);
+};
+
+export function createPaymentMethod(
+	paymentFields: PaymentFields,
+	user: User,
+): Promise<PaymentMethod> {
+	switch (paymentFields.paymentType) {
+		case 'Stripe':
+			return createStripePaymentMethod(user.isTestUser, paymentFields);
+		case 'PayPal':
+			return createPayPalPaymentMethod(user.isTestUser, paymentFields);
+		case 'DirectDebit':
+			return createDirectDebitPaymentMethod(paymentFields, user);
+		case 'Existing':
+			return Promise.reject(
+				new Error(
+					'Existing payment methods should never make their way to this lambda',
+				),
+			);
+	}
+	return Promise.reject(new Error('Unknown payment method type'));
+}
+
+export function createSalesforceContactState(
+	inputState: CreatePaymentMethodState,
+	paymentMethod: PaymentMethod,
+): WrappedState<CreateSalesforceContactState> {
+	const outputState: CreateSalesforceContactState = {
+		...inputState,
+		paymentMethod,
+	};
+
+	return {
+		state: outputState,
+		error: null,
+		requestInfo: {
+			testUser: false,
+			failed: false,
+			messages: [],
+			accountExists: false,
+		},
+	};
+}
+
+const zuoraCardTypeFromStripe = (stripeCardType: string): string => {
+	switch (stripeCardType) {
+		case 'visa':
+			return 'Visa';
+		case 'mastercard':
+			return 'MasterCard';
+		case 'amex':
+			return 'AmericanExpress';
+		case 'discover':
+			return 'Discover';
+		case 'jcb':
+			return 'JCB';
+		case 'diners':
+			return 'Diners';
+		default:
+			throw new Error(`Unknown card type ${stripeCardType}`);
+	}
+};
+
+export async function createStripePaymentMethod(
+	isTestUser: boolean,
+	paymentFields: StripePaymentFields,
+): Promise<StripePaymentMethod> {
+	const stripeService = await stripeServiceHandler.getServiceForUser(
+		isTestUser,
+	);
+	const customer = await stripeService.createCustomer(
+		paymentFields.stripePublicKey,
+		paymentFields.paymentMethod,
+	);
+	const paymentMethod = await stripeService.getPaymentMethod(
+		paymentFields.stripePublicKey,
+		paymentFields.paymentMethod,
+	);
+	const card = getIfDefined(
```

Comment:
could `stripeService.getPaymentMethod` validate this for us and only return a value if it's valid?
it looks like `card` is the only field we use in `paymentMethod`